app/src/main/java/com.example.weatherapp/data/services/WeatherService.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.weatherapp.data.services
+
+import com.example.weatherapp.BuildConfig
+import io.reactivex.rxjava3.core.Observable
+import com.example.weatherapp.data.model.ForecastModel
+import com.example.weatherapp.data.mapper.WeatherMapper
+import com.example.weatherapp.data.services.api.WeatherApiProvider
+import com.example.weatherapp.data.services.builder.WeatherRequestBuilder
+
+class WeatherService {
+
+    private val api = WeatherRequestBuilder()
```

Comment:
Can we try to use these dependencies injected in the constructor instead of creating them inside the class ? we should be very careful of using the concepts of INVERSION OF CONTROL and DEPENDENCY INVERSION

Reply:
Removed creation of classes in WeatherService this variables where moved to class parameters.